### DISCUSSION TITLE
Non-Windows Ruleset support? When will this be a possibility/become available?

### DISCUSSION THREAD

**Discussion by TheFantasticLoki** [ppy/osu#19713] [Q&A]
As the title states, it's been quite a few months since last I brought this up and I am very curious if there has been any progress made in this direction at all. Is this a proper channel/way to ask about this topic. Will it ever be possible in the first place with the use of Dynamic Link Libraries?

**Comment by Theighlin** [comment]
The path forward is being able to browse and download third party rulesets from inside the game iirc, likely with a disclaimer (all platforms). I don't think there is a proper channel to discuss this, i thought there was an issue about it but couldn't find it on a quick check. I can't say but i don't think progress has been made, this is very low priority.

**Reply by TheFantasticLoki** [reply]
I do recall there being discussion of an official ruleset listing in game but for the time being rurusetto has been covering that area for MONTHS and it already works in game as intended. I just want the ability to load my rulesets on my phone now especially since I am leaving for the weekend and std on mobile is still really cancer.

I think it's especially important to at least expose an option to enable it in the experimental/debug options as this would allow all the ruleset creators that already have gamemodes nearly fully developed and fleshed out to do the testing and development required to get their rulesets properly functional on mobile and other platforms.

**Reply by peppy** [reply]
It's literally impossible. Mobile platforms do not support dynamically loading code.

As @Theighlin mentioned, it would be enabled by us distributing the rulesets along with the game, which is far off.

If a developer wants to test on mobile, they can do so. It's very simple to deploy a build with a ruleset added.

**Reply by TheFantasticLoki** [reply]
What about using elf32 shared objects as well as dll files? As far as I can find online compiling it as a shared object is the linux equivalent to a dll file. While I doubt this would work for iOS devices  wouldn't this allow ruleset loading on linux devices and therefore also android since android is technically just another distro of linux?  Would shared object file loading have to be coded into lazer before this would work or can this work out of the box? 

EDIT TO NOT CAUSE NOTI CLUTTER: Would building in .net 6.0 allow dll loading as .net 6.0 appears to be focused on mobile application development? I think this may also help with another issue I saw about tablet driver not working on android since it requires .net 5+. Would this indeed help with that as well or is it a mute point.
